staging/src/k8s.io/client-go/tools/cache/controller.go

```diff
@@ -437,13 +440,20 @@ func newInformer(
 	})
 
 	cfg := &Config{
-		Queue:            fifo,
-		ListerWatcher:    lw,
-		ObjectType:       objType,
-		FullResyncPeriod: resyncPeriod,
-		RetryOnError:     false,
+		Queue:             fifo,
+		ListerWatcher:     lw,
+		ObjectType:        objType,
+		FullResyncPeriod:  resyncPeriod,
+		RetryOnError:      false,
+		WatchErrorHandler: watchErrorHandler,
+		ShouldResync:      shouldResync,
 
 		Process: func(obj interface{}) error {
+			if mutex != nil {
```

Comment:
Instead of this, maybe we should just create a private

`processObject` top level function that is doing what it is currently doing (without your change) and use it here


Once done, you don't need to change anything else in this file. The boilerplate of creating a config is fine to be copied in the shared_informer. So it remains as it is now, with the only change of HandleDeltas calling this new processObject function.

Reply:
@wojtek-t could you explain the benefit of this alternative way? I have no problem changing the PR, I would just like to understand.

Reply:
It's subjective, but the more vars/params/etc. I have, the harder it is to follow them and ensure they work fine.
So I personally prefer to share the code that is complex, but sharding boilerplate (e.g. setting fields in the config) I don't think it brings much value. And complicates the code that was simpler before because wasn't using all of that.

Reply:
@wojtek-t The commit has been updated with your suggestion